notebook/services/contents/filemanager.py

```diff
@@ -334,12 +334,18 @@ def _dir_model(self, path, content=True):
                     self.log.debug("%s not a regular file", os_path)
                     continue
 
-                if self.should_list(name):
-                    if self.allow_hidden or not is_file_hidden(os_path, stat_res=st):
-                        contents.append(
-                                self.get(path='%s/%s' % (path, name), content=False)
-                        )
-
+                try:
+                    if self.should_list(name):
+                        if self.allow_hidden or not is_file_hidden(os_path, stat_res=st):
+                            contents.append(
+                                    self.get(path='%s/%s' % (path, name), content=False)
+                            )
+                except OSError as e:
+                    # Ignore recursive links and move on
+                    if e.errno==62 or e.errno==40:
```

Comment:
Can we use the constants from the [errno module](https://docs.python.org/3/library/errno.html) rather than hardcoded numbers?

Reply:
Good point.  Let me look.